### PR TITLE
feat: [IDP-1440][IDP-1442]: Audit trial fixes

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57040)

### CODE REVIEW

**Comment by devesh-harness** [comment]
trigger messagecheck

**Comment by devesh-harness** [comment]
trigger messagecheck

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 5 of 6 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57040)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger uts

**Review by vikyathharekal — COMMENTED** [review]
> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57040)* status: 12 file(s) to review. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/audittrails/eventhandlers/LayoutEventHandler.java` line 69 at r5](https://reviewable.io/reviews/harness/harness-core/57040#-NmCCCp76vFHl6nssp6i:-NmCCCp76vFHl6nssp6j:b5lakcp) ([raw file](https://github.com/harness/harness-core/blob/f644e7d7e812db5340071e2dd995a86102faaae4/idp-service/src/main/java/io/harness/idp/audittrails/eventhandlers/LayoutEventHandler.java#L69)):*
> ```Java
>     LayoutUpdateEvent layoutUpdateEvent = objectMapper.readValue(outboxEvent.getEventData(), LayoutUpdateEvent.class);
> 
>     JsonNode newLayoutNode = ConfigManagerUtils.asJsonNode(layoutUpdateEvent.getNewLayout().getYaml());
> ```

nit: We can move this to a common package if we are using it outside of Config manager.



_Code quote:_
```Java
ConfigManagerUtils
```
___
*[`idp-service/src/main/java/io/harness/idp/proxy/layout/resource/LayoutProxyApiImpl.java` line 44 at r5](https://reviewable.io/reviews/harness/harness-core/57040#-NmCCL7nE_KW7wtwmDOS:-NmCCL7nE_KW7wtwmDOT:b-dzpazz) ([raw file](https://github.com/harness/harness-core/blob/f644e7d7e812db5340071e2dd995a86102faaae4/idp-service/src/main/java/io/harness/idp/proxy/layout/resource/LayoutProxyApiImpl.java#L44)):*
> ```Java
> 
>   @Override
>   //  @NGAccessControlCheck(resourceType = IDP_RESOURCE_TYPE, permission = IDP_PERMISSION)
> ```

Testing changes?



_Code quote:_
```Java
//
```

**Review by vikyathharekal — DISMISSED** [review]
> Reviewed 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57040)* status: 12 file(s) to review. 1 of 2 approvals obtained.

File: idp-service/src/main/java/io/harness/idp/audittrails/eventhandlers/LayoutEventHandler.java [hunk 1]
```diff
@@ -87,17 +92,24 @@ private boolean handleLayoutCreateEvent(OutboxEvent outboxEvent) throws IOExcept
...
+                                .timestamp(outboxEvent.getCreatedAt())
+                                .resource(ResourceDTO.fromResource(outboxEvent.getResource()))
+                                .resourceScope(ResourceScopeDTO.fromResourceScope(outboxEvent.getResourceScope()))
+                                .insertId(outboxEvent.getId())
+                                .build();
     return auditClientService.publishAudit(auditEntry, globalContext);
   }
+
+  private String removeUnwantedCharactersFromYaml(String yamlString) {
+    yamlString = yamlString.replaceFirst("---\n", "");
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/java/io/harness/idp/audittrails/eventhandlers/LayoutEventHandler.java:112]: Do we need this ? Even with this, it is still a valid YAML

File: idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java [hunk 2]
```diff
@@ -67,6 +67,7 @@ public class ConfigEnvVariablesServiceImpl implements ConfigEnvVariablesService
       "%s - is already used in plugin - %s , please use some other env variable name";
 
   private static final String ENV_VARIABLE_NOT_CONFIGURED_ERROR_MESSAGE = "Please set value for %s";
+  private static final String PREFIX_FOR_ACCOUNT = "account.";
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java:70]: We already have similar thing in GitIntegrationConstants. We can move that to common package and use it across.

File: idp-service/src/main/java/io/harness/idp/audittrails/eventhandlers/LayoutEventHandler.java (hunk 1 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/audittrails/eventhandlers/LayoutEventHandler.java:112]: yes this is for showing on UI , so just wanted to make it consistent with other audit trails on UI.

File: idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java (hunk 2 above)
Comment (devesh-harness) [reply idp-service/src/main/java/io/harness/idp/configmanager/service/ConfigEnvVariablesServiceImpl.java]: done

**Review by sathish-soundarapandian — DISMISSED** [review]

**Comment by devesh-harness** [comment]
trigger uts

**Review by vikyathharekal — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 1 files at r7, 7 of 7 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/57040#-:-NmF-MG85lbQVwml5Cuy:bnfp4nl)* status: 18 file(s) to review. 2. unresolved discussions.

**Comment by devesh-harness** [comment]
trigger ut9

**Comment by devesh-harness** [comment]
trigger ut5, ut7, ut8

**Comment by devesh-harness** [comment]
trigger ut0, ut9

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut8

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut5

**Pull request closed without merging**